src/serverrpc.cpp

```diff
@@ -0,0 +1,145 @@
+                    { "jitterBufferSize", veciJitBufNumFrames[i] },
+                    { "channels", pServer->GetClientNumAudioChannels ( i ) },
+                };
+                clients.append ( client );
+            }
+        }
+
+        // create recorder state object
+        QJsonObject recorder{
+            { "initialised", pServer->GetRecorderInitialised() },
+            { "errorMessage", pServer->GetRecorderErrMsg() },
+            { "enabled", pServer->GetRecordingEnabled() },
+            { "recordingDirectory", pServer->GetRecordingDir() },
+        };
+
+        // create result object
+        QJsonObject result{
+            { "clients", clients },
+            { "recorder", recorder },
+        };
+        response["result"] = result;
+        Q_UNUSED ( params );
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusserver/getServerProfile", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QJsonObject result{
+            { "name", pServer->GetServerName() },
+            { "city", pServer->GetServerCity() },
+            { "countryId", pServer->GetServerCountry() },
+            { "welcomeMessage", pServer->GetWelcomeMessage() },
+            { "registrationStatus", pServer->GetSvrRegStatus() },
+        };
+        response["result"] = result;
+        Q_UNUSED ( params );
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusserver/setServerName", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        auto jsonServerName = params["serverName"];
+        if ( !jsonServerName.isString() )
+        {
+            response["error"] = CRpcServer::CreateJsonRpcError ( CRpcServer::iErrInvalidParams, "Invalid params: serverName is not a string" );
+            return;
+        }
+
+        pServer->SetServerName ( jsonServerName.toString() );
+        response["result"] = "ok";
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusserver/setWelcomeMessage", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        auto jsonWelcomeMessage = params["welcomeMessage"];
+        if ( !jsonWelcomeMessage.isString() )
+        {
+            response["error"] = CRpcServer::CreateJsonRpcError ( CRpcServer::iErrInvalidParams, "Invalid params: welcomeMessage is not a string" );
+            return;
+        }
+
+        pServer->SetWelcomeMessage ( jsonWelcomeMessage.toString() );
+        response["result"] = "ok";
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusserver/setRecordingDirectory", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        auto jsonRecordingDirectory = params["recordingDirectory"];
+        if ( !jsonRecordingDirectory.isString() )
+        {
+            response["error"] =
+                CRpcServer::CreateJsonRpcError ( CRpcServer::iErrInvalidParams, "Invalid params: jsonRecordingDirectory is not a string" );
+            return;
+        }
+
+        pServer->SetRecordingDir ( jsonRecordingDirectory.toString() );
```

Comment:
IIRC, when it starts recording, the recorder tries to create the directory using a Qt call that claims to create the path.  Of course, it needs the permission.

`main.cpp` does a lot of checking up front, so that any argument passed in to the server is "as valid as possible".

Reply:
It seems to do for the persistence file. But I don't see it in main.cpp for the recorder - only the persistence file. But the UI doesn't seem to do anything different either.

Reply:
In that case, it's probably checking as much as the UI checked -- it leaves most of the work to the recorder (which may just store the string without complaint until it starts, I can't remember).

Reply:
Here’s what will happen:

1. The RPC will return "acknowledged"
2. Client has to call `jamulusserver/getServerStatus` to check if the recording is actually enabled. If not the error message will be shown in `result.recorder.errorMessage`.